nf.js

```diff
@@ -43,12 +44,68 @@ var calculatePadding = _requirements.calculatePadding;
 var startProxies = require('./lib/proxy').startProxies;
 var startForward = require('./lib/forward').startForward;
 
+var pidPath = path.resolve(process.env.PWD, './.nf.run');
+
 // Kill All Child Processes on SIGINT
 process.once('SIGINT', function() {
   display.Warn('Interrupted by User');
+  fs.unlink(pidPath);
   emitter.emit('killall', 'SIGINT');
 });
 
+process.on('SIGHUP', function() {
+  display.Alert('Restart by User');
+  emitter.emit('restart', 'SIGINT');
+  kickstart();
+});
+
+fs.open(pidPath, 'r', function(err) {
```

Comment:
It looks like you are trying to avoid the `fs.stat`+`fs.open` race condition by replacing it with an `fs.open`+`fs.writeFile` race condition with a bonus of file descriptor leak.